server/src/main/java/org/opensearch/search/SearchBootstrapSettings.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search;
+
+import org.opensearch.common.settings.Setting;
+import org.opensearch.common.settings.Settings;
+
+/**
+ * Keeps track of all the search related node level settings which can be accessed via static methods
+ */
+public class SearchBootstrapSettings {
+    // settings to configure maximum slice created per search request using OS custom slice computation mechanism. Default lucene
+    // mechanism will not be used if this setting is set with value > 0
+    public static final String CONCURRENT_SEGMENT_SEARCH_TARGET_MAX_SLICE_COUNT_KEY = "search.concurrent.max_slice";
+    public static final int CONCURRENT_SEGMENT_SEARCH_TARGET_MAX_SLICE_COUNT_DEFAULT_VALUE = -1;
+
+    // value <= 0 means lucene slice computation will be used
+    public static final Setting<Integer> CONCURRENT_SEGMENT_SEARCH_TARGET_MAX_SLICE_COUNT_SETTING = Setting.intSetting(
+        CONCURRENT_SEGMENT_SEARCH_TARGET_MAX_SLICE_COUNT_KEY,
+        CONCURRENT_SEGMENT_SEARCH_TARGET_MAX_SLICE_COUNT_DEFAULT_VALUE,
+        Setting.Property.NodeScope
+    );
+    private static Settings settings;
+
+    public static void initialize(Settings openSearchSettings) {
+        settings = openSearchSettings;
+    }
+
+    public static int getValueAsInt(String settingName, int defaultValue) {
```

Comment:
This is a bit weird because any setting can be passed in here even if unrelated to search. I'd probably implement this as `SearchBootstrapSettings.getTargetMaxSliceCount()`. I'd consider using `Optional<Integer>` or a nullable `Integer` as the return type as opposed to using the magic value of -1.

Reply:
1. I thought about adding the `getTargetMaxSliceCount` method but then it will not keep this class generic which can be utilized for other search settings. For each new setting we will need to add specific method. But I see your point about being able to access any settings. But that is true for any access to `ClusterSettings` from different components as well. Let me know if you still prefer adding explicit method `getTargetMaxSliceCount` here.
2. -1 is the default value for this setting as it doesn't allow `null` for it. When the setting will be converted to dynamic then it will return the `default` value when setting value is looked up using `clusterSettings.get(CONCURRENT_SEGMENT_SEARCH_TARGET_MAX_SLICE_COUNT_SETTING)` and it is not explicitly set. It will not return null so caller needs to handle this default of `-1` to fallback to lucene behavior in that case. Hence keeping it as is without adding nullability check for now. The cases will be:

* Not set explicitly --> default value will be returned so use lucene slice computation
* Set explicitly to -1/0 --> use lucene slice computation
* Set explicitly to >0 --> use custom slice computation